core/src/main/scala/kafka/log/LogConfig.scala

```diff
@@ -256,7 +256,7 @@ object LogConfig {
         MEDIUM, CompressionTypeDoc, KafkaConfig.CompressionTypeProp)
       .define(PreAllocateEnableProp, BOOLEAN, Defaults.PreAllocateEnable, MEDIUM, PreAllocateEnableDoc,
         KafkaConfig.LogPreAllocateProp)
-      .define(MessageFormatVersionProp, STRING, Defaults.MessageFormatVersion, in(ApiVersion.allValidVersions.toArray:_*), MEDIUM, MessageFormatVersionDoc,
+      .define(MessageFormatVersionProp, STRING, Defaults.MessageFormatVersion, ValidPrefixString.in(ApiVersion.allValidVersions.toArray:_*), MEDIUM, MessageFormatVersionDoc,
```

Comment:
Since we only have on use case for this at the moment for validating ApiVersion, I wonder if it would be simpler to have a custom `ApiVersion.Validator` which invokes `ApiVersion.apply` directly? My concern is that we have two separate paths for validation. The logic seems consistent as far as I can tell, but we may change the validation logic in the future.

Reply:
Correct me if I am wrong. Currently, `Validator` and its implementations are in `clients` codebase which does not see any classes from within the `core`.  It cannot invoke `ApiVersion.apply`. Right?

Reply:
To clarify, I am suggesting that we add a custom `Validator` implementation inside `ApiVersion`. We need not put it in the clients module since `ApiVersion` is in core.